app/backend/src/clj/teet/land/owner_opinion_export.clj

```diff
@@ -14,14 +17,133 @@
+    (if (= amount 1)                                        ;; Don't use ordered list if only 1 opinion
+      (md/render-md-html (:land-owner-opinion/authority-position
+                           (first opinions)))
+      [:ol
+       (for [opinion opinions]
+         [:li
+          (if-let [position (:land-owner-opinion/authority-position
+                              opinion)]
+            (md/render-md-html position)
+            [:p
+             [:i (tr* :no-position-added)]])])])))
+
+(defn unit-opinions-table
+  [estates]
+  [:div
+   [:table {:border "1" :cellspacing "0" :cellpadding "5"}
+    [:thead
+     [:th (tr* :jrk)]
+     [:th (tr* :estate-num-name)]
+     [:th (tr* :cadastral-id-name)]
+     [:th (tr* :opinion-body)]
+     [:th (tr* :authority-position)]]
+    [:tbody
+     (doall
+       (map-indexed
+         (fn [estate-idx [estate-id units]]
+           (let [unit-count (count units)]
+             (map-indexed
+               (fn [unit-idx {:keys [L_AADRESS TUNNUS teet-opinions] :as unit}]
+                 [:tr
+                  [:td {:style "vertical-align: top"}
+                   [:p (jrk-nr estate-idx unit-idx unit-count)]]
+                  [:td {:style "vertical-align: top"}
+                   [:p estate-id]]
+                  [:td {:style "vertical-align: top"}
+                   [:p (str TUNNUS ";" L_AADRESS)]]
+                  [:td {:style "vertical-align: top"}
+                   (render-opinions teet-opinions)]
+                  [:td {:style "vertical-align: top"}
+                   (render-positions teet-opinions)]])
+               units)))
+         estates))]]])
+
+(defn units-without-opinions-table
+  [estates]
+  [:div
+   [:table {:border "1" :cellspacing "0" :cellpadding "5"}
+    [:thead
+     [:th (tr* :jrk)]
+     [:th (tr* :estate-num-name)]
+     [:th (tr* :cadastral-id-name)]]
+    [:tbody
+     (doall
+       (map-indexed
+         (fn [estate-idx [estate-id units]]
+           (let [unit-count (count units)]
+             (map-indexed
+               (fn [unit-idx {:keys [L_AADRESS TUNNUS]}]
+                 [:tr
+                  [:td
+                   [:p (jrk-nr estate-idx unit-idx unit-count)]]
+                  [:td
+                   [:p estate-id]]
+                  [:td
+                   [:p (str TUNNUS ";" L_AADRESS)]]])
+               units)))
+         estates))]]])
+
+(defn summary-table [db activity opinion-type {:keys [api-url api-secret] :as _config}]
+  (let [opinions (opinions-by-type db activity opinion-type)
+        project (ffirst
+                  (d/q '[:find (pull ?project [:thk.project/name :thk.project/project-name
+                                               :db/id :thk.project/id])
+                         :in $ ?activity
+                         :where
+                         [?lifecycle :thk.lifecycle/activities ?activity]
+                         [?project :thk.project/lifecycles ?lifecycle]]
+                       db activity))
```

Comment:
This code is essentially a function from project namespace and can be refactored as such.